lib/test_prof/recipes/rspec/let_it_be.rb

```diff
@@ -114,6 +113,79 @@ def define_let_it_be_methods(identifier, **modifiers)
 
       let(identifier, &let_accessor)
     end
+
+    def define_freezing_hooks
+      # Prevent hooks from being defined several times
+      return if instance_variable_get(:"#{PREFIX}hooks_defined")
+
+      before(:all) do
+        let_it_be_objects = instance_variable_get(:"#{TestProf::LetItBe::PREFIX}let_it_be_objects")
+
+        let_it_be_objects.each { |object| Freezer.deep_freeze(object) }
+      end
+
+      prepend_after do |example|
+        if example.exception&.message&.match?(/can't modify frozen Hash/)
+          example.exception.message << FROZEN_HASH_HINT
+        end
+      end
+
+      instance_variable_set(:"#{PREFIX}hooks_defined", true)
+    end
+
+    # Exception needs to be handled both here and in `handle_frozen_hash_error`
+    # because if it is raised in before_all it isn't caught in `after` block and
+    # if it's inside the example it isn't raised so it has to be handled in `after`.
+    def build_let_it_be_initializer(identifier, freeze, &block)
+      proc do
+        begin
+          record = instance_exec(&block)
+          if freeze
+            # FIXME: When models and their associations are defined with different
+            # options, e.g. `reload: true` and without it, the ones that are not
+            # supposed to be frozen will still be frozen here. Add those with
+            # `reload: true`/`refind: true`/`freeze: false` to ignore list.
+            let_it_be_objects = instance_variable_get(:"#{TestProf::LetItBe::PREFIX}let_it_be_objects")
+            let_it_be_objects ||= instance_variable_set(:"#{TestProf::LetItBe::PREFIX}let_it_be_objects", [])
+            let_it_be_objects << record
+          end
+
+          instance_variable_set(:"#{TestProf::LetItBe::PREFIX}#{identifier}", record)
+        rescue => e
```

Comment:
I've modified this to rescue from `FrozenError`, but it's not in Ruby 2.4.
Wondering how [this](https://github.com/palkan/test-prof/pull/178/files#diff-cb255553719186df25a3969e3f716618R18) doesn't fail:
```ruby
expect { ... }.to raise_error(FrozenError, /can't modify frozen Hash/)
```